### PR TITLE
ENH: Update vxl for export all symbols

### PR DESCRIPTION
**Pull request by hjmjohnson**
Using VXL 2.0.0 version that allows for exporting all symbols in windows shared library builds.

### CODE REVIEW

File: Modules/ThirdParty/VNLInstantiation/src/libraryplaceholder.cxx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+/* This file has one symbol that is exported in order to create a valid
+ * library for backwards compatibility in linking applications that may
+ * depend on the existence of the VNLInstantiation library
+ */
+
+extern bool dummy_placeholder_variable_to_avoid_empty_library_warnings;
```
Comment (dzenanz) [inline Modules/ThirdParty/VNLInstantiation/src/libraryplaceholder.cxx]: Wouldn't this create undefined symbol for `dummy_...` if/when somebody does link to `ITKVNLInstantiation`?

**Comment by dzenanz** [comment]
ghostflow fails due to trailing whitespace. Click on [details](https://github.com/InsightSoftwareConsortium/ITK/pull/175/checks?check_run_id=31458142).

File: Modules/ThirdParty/VNLInstantiation/src/libraryplaceholder.cxx (hunk 1 above)
Comment (hjmjohnson) [reply Modules/ThirdParty/VNLInstantiation/src/libraryplaceholder.cxx]: If we only had an extern declaration then you would be correct, but line 8 defines the symbol.

**Comment by hjmjohnson** [comment]
@dzenanz an update that removes end-of-line spacing will be coming as soon as the VXL robots verify the fix in the upstream version :  https://github.com/vxl/vxl/pull/546

**Comment by hjmjohnson** [comment]
> ghostflow fails due to trailing whitespace. Click on [details](https://github.com/InsightSoftwareConsortium/ITK/pull/175/checks?check_run_id=31458142).

After several VXL update rounds, I think we have fixed all the trailing whitespace issues.

**Comment by hjmjohnson** [comment]
@dzenanz Would you mind testing to see if this fixes the ITK build failure?

**Comment by dzenanz** [comment]
The failure was occurring on 3 nightly builds, on of which is "Win-VS17-Visual Studio 15 2017 Win64-Release-Shared". It should build overnight and report to the [dashboard](https://open.cdash.org/index.php?project=Insight#!#Expected_Nightly).

**Comment by hjmjohnson** [comment]
@thewtex @dzenanz It looks like the vxl errors are gone, and now other errors have emerged.

**Comment by dzenanz** [comment]
One of the dashboard machines restarted and didn't execute the builds, the other has some configuration issue (rc.exe not available in path). I started KRSDash-2084, so the results should be ready tomorrow.

**Comment by dzenanz** [comment]
This fixed the build errors coming from VXL. There are other errors now, e.g. `ITKAnisotropicSmoothingTestDriver.obj : error LNK2019: unresolved external symbol "struct RegressionTestParameters regressionTestParameters" (?regressionTestParameters@@3URegressionTestParameters@@A) referenced in function main [C:\Jenkins\workspace\ITKWinVS17\ITK-bin\Modules\Filtering\AnisotropicSmoothing\test\ITKAnisotropicSmoothingTestDriver.vcxproj]`. I remember trying to fix this error, which ended in miserable failure to diagnose the source. Now that I know what is causing it, I should be able to fix it 😄 #191

**Pull request merged**